frontend/src/components/Dashboard.jsx

```diff
@@ -86,546 +86,580 @@ const DashBoardPage = styled.div`
-  const [supportedNetworks, setSupportedNetworks] = useState();
-  const [isLoading, setIsLoading] = useState(false);
-  const [paymasterBalance, setPaymasterBalance] = useState("0");
-  const [useCustomPaymaster] = useState(false);
-  const [customPaymasterAddress] = useState();
-  const [selectedOption, setSelectedOption] = useState(0);
-  const [amount, setAmount] = useState(0);
-  const [checked, setChecked] = useState(false);
-  const [buttonText, setButtonText] = useState("Deposit");
-  const [whiteListAddress, setWhitelistAddress] = useState("");
+	// Definitions
+	const { user, signIn } = UserAuth();
+	const [value, setValue] = React.useState(0);
+	const [chainId, setChainId] = useState(80001);
+	const [signedIn, setSignedIn] = useState(false);
+	const [networksSupported, setNetworksSupported] = useState(
+		Object.keys(networks)
+	);
+	const [supportedNetworks, setSupportedNetworks] = useState();
+	const [isLoading, setIsLoading] = useState(false);
+	const [paymasterBalance, setPaymasterBalance] = useState("0");
+	const [useCustomPaymaster] = useState(false);
+	const [customPaymasterAddress] = useState();
+	const [selectedOption, setSelectedOption] = useState(0);
+	const [amount, setAmount] = useState(0);
+	const [checked, setChecked] = useState(false);
+	const [buttonText, setButtonText] = useState("Deposit");
+	const [whiteListAddress, setWhitelistAddress] = useState("");
+	const [removeWhitelist, setRemoveWhitelist] = useState(false);
+	const [whitelistButtonText, setWhitelistButtonText] =
+		useState("Add to Whitelist");
 
-  // Functions
-  const getPaymasterContract = (chainId) => {
-    if (!supportedNetworks) return null;
-    const provider = new ethers.providers.JsonRpcProvider(
-      supportedNetworks[chainId].rpcUrl,
-      {
-        name: "Connected Bundler",
-        chainId: Number(chainId),
-      }
-    );
-    if (useCustomPaymaster) {
-      return new ethers.Contract(
-        customPaymasterAddress,
-        EtherspotPaymasterAbi,
-        provider
-      );
-    } else {
-      return new ethers.Contract(
-        supportedNetworks[chainId].paymasterAddress,
-        EtherspotPaymasterAbi,
-        provider
-      );
-    }
-  };
+	// Functions
+	const getPaymasterContract = (chainId) => {
+		if (!supportedNetworks) return null;
+		const provider = new ethers.providers.JsonRpcProvider(
+			supportedNetworks[chainId].rpcUrl,
+			{
+				name: "Connected Bundler",
+				chainId: Number(chainId),
+			}
+		);
+		if (useCustomPaymaster) {
+			return new ethers.Contract(
+				customPaymasterAddress,
+				EtherspotPaymasterAbi,
+				provider
+			);
+		} else {
+			return new ethers.Contract(
+				supportedNetworks[chainId].paymasterAddress,
+				EtherspotPaymasterAbi,
+				provider
+			);
+		}
+	};
 
-  const fetchData = async (address) => {
+	const fetchData = async (address) => {
 		try {
 			setIsLoading(true);
 			const data = await (
 				await fetch("http://localhost:5050/getSupportedNetworks", {
```

Comment:
Should this not be an env var?

Reply:
I have addressed that on earlier PR to add it to constants file